### PR TITLE
Monitor model improvements

### PR DESCRIPTION
**Pull request by bevans2000**
Objective of this PR is to addresses performance and code smells.

- Remove a lot of duplicate code by creating a ```AbstractMonitorModel``` class to hold common methods
- Create a ```ColumnSpec``` record that simplifies the definition of columns by tying name & class.
- In ```VehicleTableModel``` and ``SettlementTableModel``` change how resource driven column and updated/extracted. These now use a mapping from the resource to the column index

### CODE REVIEW

File: mars-sim-ui/src/main/java/org/mars_sim/msp/ui/swing/tool/monitor/AbstractMonitorModel.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+/**
+ * Default implementation of the MonitorModel
+ */
+public abstract class AbstractMonitorModel extends AbstractTableModel
+        implements MonitorModel {
+
+    /**
+     * Helper class to define the specification of a column
+     */
+    protected record ColumnSpec (String name, Class<?> type) implements Serializable {};
```
Comment (mokun) [inline mars-sim-ui/src/main/java/org/mars_sim/msp/ui/swing/tool/monitor/AbstractMonitorModel.java:24]: awesome that you are using `record` now.

**Comment by mokun** [comment]
One question is, how do we make Monitor Tool compatible with our future plan of adding a sponsor combobox ? 

See [this]( https://github.com/mars-sim/mars-sim/issues/903).

File: mars-sim-ui/src/main/java/org/mars_sim/msp/ui/swing/tool/monitor/AbstractMonitorModel.java (hunk 1 above)
Comment (bevans2000) [reply mars-sim-ui/src/main/java/org/mars_sim/msp/ui/swing/tool/monitor/AbstractMonitorModel.java:24]: I have started using them elsewhere as well. An advantage of upgrading the Java version

**Comment by bevans2000** [comment]
It can be done. Needs some planning and consideration but the MonitorModel will filter of a Set of Settlements instead of one. 
All models then need to be clever to automatically add a Settlement column if more than one is selected.
But I still think the direction of the ui in terms of layout needs considering, #979

**Pull request merged**